### PR TITLE
GEODE-6252: Clean up dependency management

### PR DESCRIPTION
**Pull request by PurelyApplied**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [kinda] Have you written or updated unit tests to verify your changes? [This change set has left the produced poms unchanged.]

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: gradle/geode-dependency-management.gradle [hunk 1]
```diff
@@ -18,146 +18,245 @@
...
+      }
+    }
+  }
+
+  void set_extended_property_versioning(Project project) {
+    // These versions are consumed beyond the scope of source set dependencies.
+    project.ext {
+      // These version numbers are consumed by :geode-modules-assembly:distAppServer filtering
+      // Some of these are referenced above as well
+      set('antlr.version', '2.7.7')
```
Comment (rhoughton-pivot) [inline gradle/geode-dependency-management.gradle]: I like this much better than the plain-text file holding the name-value pairs of package to version.

File: gradle/geode-dependency-management.gradle [hunk 2]
```diff
@@ -18,146 +18,245 @@
...
+          entry('geode-protobuf-messages')
+          entry('geode-pulse')
+          entry('geode-rebalancer')
+          entry('geode-wan')
+          entry('geode-web')
+          entry('geode-web-api')
+        }
+
+        // informal, inter-group dependencySet
+        dependency('com.sun.xml.bind:jaxb-impl:2.3.1')
```
Comment (rhoughton-pivot) [inline gradle/geode-dependency-management.gradle]: Do we know if these versions are actually coupled?

Comment (PurelyApplied) [reply gradle/geode-dependency-management.gradle]: Dunno!  I just noticed they're both JAXB and on the same version.  I'll dig into it.

Comment (PurelyApplied) [reply gradle/geode-dependency-management.gradle]: As best as I can tell, this is just pathological out in Maven.

[jaxb-impl](https://mvnrepository.com/artifact/com.sun.xml.bind/jaxb-impl/2.3.1) from Sun is the up-to-date version of the impl, but [jaxb-api](https://mvnrepository.com/artifact/javax.xml.bind/jaxb-api/2.3.1) from jaxb.xml.bind is the up-to-date api.  Everything else similarly named appears to have been abandoned in 2010.

**Pull request merged**